plugin/storage/integration/domain_trace_compare_test.go

```diff
@@ -0,0 +1,138 @@
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/uber/jaeger/model"
+)
+
+type TraceByKey []*model.Trace
+
+func (s TraceByKey) Len() int      { return len(s) }
+func (s TraceByKey) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s TraceByKey) Less(i, j int) bool {
+	if len(s[i].Spans) != len(s[j].Spans) {
+		return len(s[i].Spans) < len(s[j].Spans)
+	} else if len(s[i].Spans) == 0 {
+		return true
+	}
+	return s[i].Spans[0].TraceID.Low < s[j].Spans[0].TraceID.Low
+}
+
+func CompareListOfModelTraces(t *testing.T, expected []*model.Trace, actual []*model.Trace) {
+	sort.Sort(TraceByKey(expected))
+	sort.Sort(TraceByKey(actual))
+	assert.Equal(t, len(expected), len(actual))
+	for i := range expected {
+		CompareModelTraces(t, expected[i], actual[i])
+	}
+}
+
+type SpanByKey []*model.Span
+
+func (s SpanByKey) Len() int           { return len(s) }
+func (s SpanByKey) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+func (s SpanByKey) Less(i, j int) bool { return s[i].SpanID < s[j].SpanID }
+
+func CompareModelTraces(t *testing.T, expected *model.Trace, actual *model.Trace) {
+	if expected.Spans == nil {
+		require.Nil(t, actual.Spans)
+		return
+	}
+	checkTraceID(t, expected)
+	checkTraceID(t, actual)
+	expectedSpans := expected.Spans
+	actualSpans := actual.Spans
+	sort.Sort(SpanByKey(expectedSpans))
+	sort.Sort(SpanByKey(actualSpans))
+	assert.Equal(t, len(expectedSpans), len(actualSpans))
+	for i := range expectedSpans {
+		compareModelSpans(t, expectedSpans[i], actualSpans[i])
+	}
+}
+
+func checkTraceID(t *testing.T, trace *model.Trace) {
+	traceID := trace.Spans[0].TraceID.String()
+	for _, span := range trace.Spans {
+		assert.Equal(t, traceID, span.TraceID.String())
+	}
+}
+
+func compareModelSpans(t *testing.T, expected *model.Span, actual *model.Span) {
+	// TODO: Don't manually enumerate, because if we add a new field, this test would still pass.
+	assert.Equal(t, expected.TraceID.Low, actual.TraceID.Low)
+	assert.Equal(t, expected.TraceID.High, actual.TraceID.High)
+	assert.Equal(t, expected.SpanID, actual.SpanID)
+	assert.Equal(t, expected.OperationName, actual.OperationName)
+	assert.Equal(t, expected.Flags, actual.Flags)
+	assert.Equal(t, expected.StartTime.UnixNano()/1000, actual.StartTime.UnixNano()/1000)
+	assert.Equal(t, expected.Duration/1000, actual.Duration/1000)
+	compareModelTags(t, expected.Tags, actual.Tags)
+	compareModelLogs(t, expected.Logs, actual.Logs)
+	compareModelProcess(t, expected.Process, actual.Process)
+}
+
+type TagByKey []model.KeyValue
+
+func (t TagByKey) Len() int           { return len(t) }
+func (t TagByKey) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
+func (t TagByKey) Less(i, j int) bool { return t[i].Key < t[j].Key }
+
+func compareModelTags(t *testing.T, expected []model.KeyValue, actual []model.KeyValue) {
+	sort.Sort(TagByKey(expected))
+	sort.Sort(TagByKey(actual))
```

Comment:
not necessarily. First, you can use `pretty` to print the diffs, as done [here](model/converter/thrift/jaeger/to_domain_test.go:56). Another trick I used was to write out the JSON to files (when the test fails) so that it can be compared with other tools like `diff`.